backend/tracim_backend/lib/utils/app.py

```diff
@@ -0,0 +1,216 @@
+        :param fa_icon: font-awesome icon associated to the content-type, ex: paperclip
+        :param label: label of the content-type, ex: "File"
+        :param creation_label: label for creation, ex: "Upload a file".
+        :param available_statuses: status available for this content type
+        :param app: associated app
+        :param slug_aliases: list of alias slug which refer to same content-type
+        :param allow_sub_content: should be true for folder-like content type, false for other
+        :param file_extension: default file extension of content_type. ex: ".document.html"
+        :param minimal_role_content_creation: minimal workspace role needed for create content of
+        this kind.
+        """
+        self.slug = slug
+        self.fa_icon = fa_icon
+        self.label = label
+        self.creation_label = creation_label
+        self.available_statuses = available_statuses
+        self.slug_aliases = slug_aliases
+        self.allow_sub_content = allow_sub_content
+        self.file_extension = file_extension
+        self.minimal_role_content_creation = minimal_role_content_creation
+        self.app = app
+
+
+class TracimApplication(ABC):
+    """
+    Specification for Tracim Application
+
+    Create a subclass of this application to create a new application and load it into tracim.
+    Default tracim app are loaded by importing "tracim_backend/applications/<application_name>/application.py"
+    and running local create_app() method which return a TracimApplication.
+    you need to implements all the methods of TracimApplication, to make application work correctly.
+
+    Use "pass" if you don't want to put anything in one of the required method. example:
+
+    >>> def load_config(self, app_config: "CFG") -> None:
+    ...    pass
+    ... # doctest: +SKIP
+
+    If you want more examples, check "tracim_backend.applications.<application_name>.application" files.
+    """
+
+    def __init__(
+        self, label: str, slug: str, fa_icon: str, config: typing.Dict[str, str], main_route: str
+    ) -> None:
+        """
+
+        :param label: label of the application, ex: "Files"
+        :param slug: slug of the application, ex: "contents/file"
+        :param fa_icon: font-awesome icon associated to the application, ex: paperclip
+        :param config: specific config of the app.
+        :param main_route: route used in app sidebar entry, allow "workspace_id" as variable, ex:
+         "/ui/workspaces/{workspace_id}/contents?type=file",
+        """
+        self.label = label
+        self.slug = slug
+        self.fa_icon = fa_icon
+        self.config = config
+        self.main_route = main_route
+        self.content_types = []  # typing.List[ContentType]
+        # INFO - G.M - 2020-01-16 - will be set to true automatically by Tracim
+        # if app slug in is APP_ENABLED.
+        self.is_active = False
+
+    def load_content_types(self) -> None:
```

Comment:
`abc` offer a modern way to set an method as abstract, see https://docs.python.org/3/library/abc.html#abc.abstractmethod

Reply:
done whenever possible.